docs/api.asciidoc

```diff
@@ -26,6 +26,18 @@ entirely.
 
 NOTE: You cannot access these APIs via the Console in Kibana.
 
+== Request Headers
+
+The following headers must be set on all non-whitelisted API endpoints 
+(See https://www.elastic.co/guide/en/kibana/current/kibana-authentication.html#saml[SAML Authentication]
```

Comment:
I think it'd be worthwhile to document `server.xsrf.whitelist` in the main settings page, and link there instead, with something like:
```
See `server.xsrf.whitelist` in <<settings>> for instructions on whitelisting API endpoints.
```

As an aside, we try to use AsciiDoc's linking mechanism (the `<<link-name>>` syntax in my example above) to cross-reference pages, that way the links automatically point to the correct version of the docs (`current` wouldn't be the correct version to link to when referencing a different version of the stack docs).